app/src/main/java/net/grandcentrix/blutufintegration/ui/list/ListViewModel.kt

```diff
@@ -8,27 +8,38 @@ import net.grandcentrix.blutuf.core.Blutuf
 import net.grandcentrix.blutuf.core.api.BlutufEvent
 import net.grandcentrix.blutuf.core.api.BlutufEventResult
 import net.grandcentrix.blutufintegration.data.model.DeviceUiState
-import net.grandcentrix.blutufintegration.data.repo.BluetoothRepository
+import net.grandcentrix.blutufintegration.data.model.ProcessState
+import net.grandcentrix.blutufintegration.domain.*
 
 private const val SCAN_TIMEOUT: Long = 5000
 
-class ListViewModel(private val bluetoothRepository: BluetoothRepository) : ViewModel() {
+class ListViewModel(
+    private val scanUseCase: ScanUseCase,
+    private val stopScanUseCase: StopScanUseCase,
+    private val connectUseCase: ConnectUseCase,
+    private val disconnectUseCase: DisconnectUseCase,
+    getSelectedDeviceUseCase: GetSelectedDeviceUseCase
+) : ViewModel() {
 
-    val uiModel = bluetoothRepository.devicesStateFlow.asLiveData()
-    val selectedDevice = bluetoothRepository.selectedDeviceStateFlow.asLiveData()
+    val uiModel = MutableLiveData<ProcessState<List<DeviceUiState>>>()
+    val selectedDevice = liveData {
```

Comment:
Seeing this `liveData {}` I guess this can be replaced by `Flow.asLiveData()`, right?
If not I would write your own extension function to convert a flow to LiveData.

Reply:
yess, you’re right 🙈